optuna/trial/_trial.py

```diff
@@ -469,6 +469,10 @@ def objective(trial):
         Raises:
             :exc:`NotImplementedError`:
                 If trial is being used for multi-objective optimization.
+            :exe:`ValueError`:
+                If ``step is not None`` is specified.
+            :exe:`TypeError`:
+                If ``value of type but a float`` is specified.
```

Comment:
I guess this sentence is a bit fragmented. In my understanding, the `TypeError` is raised when the type of `value` is not float-castable. Could you rephase it, for example, as follows?

```suggestion
            :exe:`TypeError`:
                If the type of ``value`` is not obj:`float`.
```

Reply:
I also think I should revise the wording that way.